---
title: "GxE Testing - PanC, T2D and RG"
author: "Edwina Rossi"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r initial setup, echo = F}
#setup
library(qqman)
library(data.table)
library(interactions)
library(jtools)
library(magrittr)
library(dplyr)
library(broom)
library(forestmangr)
library(ggpubr)
library(finalfit)
library(ggeffects)
library(lme4)
library(splines)
library(sjmisc)
library(ggplot2)

# set working directory 
setwd("~/Desktop/Masters Project/GxE Testing PanC")

# importing T2D phenotype file
pheno_file = fread("Pheno_file_RG_T2D_PANC.txt")

# adding log RG.control column 
pheno_file$RG.control.log = ifelse((!is.na(pheno_file$Random.glucose.log) & (pheno_file$T2D == 0)), pheno_file$Random.glucose.log, NA)

```


# Investigating the Association Between T2D and Risk of Pancreatic Cancer - GxE Analyses
## Case-Control Numbers and Chi-Squared Test

```{r Case & Control + Chi-Squared Test}
# case and control numbers 
with(pheno_file, addmargins(table(T2D, C25)))

# percentage terms
with(pheno_file, prop.table(table(T2D, C25), margin = 1)*100)

# Chi Squared test
with(pheno_file, chisq.test(T2D, C25))
```

## GxE Testing with SNPs
### QQ Plot 
In genetic studies, a QQ (quantile-quantile) plot is used to graphically represent the deviation of the observed P values from the null hypothesis. In simple terms, a QQ plot is a scatter plot of observed P values for each SNP (sorted from largest to smallest), plotted against expected values from a theoretical Chi-squared distribution. In the case that the observed values are equal to the expected values, all points should fall on or close to the middle line. 

```{r QQ Plot, echo = FALSE}
# importing interaction results file
interaction_results = fread("interaction_t2d-panc_2022-05-31_combined.csv")

# qq plot 
qq(interaction_results$`P-value`)

```
In the above QQ plot, we see that a relatively large proportion of points fall above the middle line. This indicates that these p values are more significant than expected under the null hypothesis. 

### GxE Testing with SNPs 
```{r GxE Testing with SNPs}

# Bonferroni correction 
n_independent_hits = 106
p_val = 0.05/n_independent_hits

# significant SNPs filtered by Bonferroni correction 
sig_snps = interaction_results %>% filter(`P-value` <= p_val)

T2D_sumstats_prefiltered = fread("screened_snps_t2d_2021-12-09.txt")
colnames(T2D_sumstats_prefiltered)[colnames(T2D_sumstats_prefiltered) == "ALLELE1"] = "EA"
colnames(T2D_sumstats_prefiltered)[colnames(T2D_sumstats_prefiltered) == "ALLELE0"] = "NEA"
colnames(T2D_sumstats_prefiltered)[colnames(T2D_sumstats_prefiltered) == "SNP"] = "rsID"

significant_snps = inner_join(sig_snps, T2D_sumstats_prefiltered, by = "rsID") %>% select(rsID, chromosome, position, EA.x, NEA.x, EA.y, NEA.y, BETA, P_BOLT_LMM_INF, `Effect size`, `P-value`)

significant_snps$temp_T2D_EA = significant_snps$EA.y
significant_snps$temp_T2D_NEA = significant_snps$NEA.y

significant_snps$EA.y = with(significant_snps, ifelse(BETA < 0, temp_T2D_NEA, temp_T2D_EA))
significant_snps$NEA.y = with(significant_snps, ifelse(BETA < 0, temp_T2D_EA, temp_T2D_NEA))

significant_snps$temp_T2D_EA = NULL
significant_snps$temp_T2D_NEA = NULL

significant_snps$BETA = significant_snps$BETA*-1

significant_snps %>% select(-EA.y, -NEA.y)

#EA.y - from T2D summary statistics
#EA.x - from significant results

```

After performing Bonferonni correction to account for the multiple testing burden, no statistically significant SNPs are identified. This is likely due to the low statistical power of the GxE analysis. 

## Exploratory Regression Analyses - T2D and PANC
This section contains some simple regression analyses to investigate the relationship between T2D and certain covariates on PANC outcome. 

### Association Between T2D and Pancreatic Cancer Without Adjustment by Covariates

```{r T2D and PanC - no covs, echo = FALSE}

# load package for OR 
#library(MASS)

# simple regression analysis between T2D and PANC 
t2d_pred_panc = glm(C25 ~ T2D, family=binomial("logit"), 
                        data = pheno_file)
summary(t2d_pred_panc)

# Odds ratios and p-value of T2D 
cat("OR = ", round(exp(coef(summary(t2d_pred_panc)))[,1][2], 2), 
    ", P-value = ", coef(summary(t2d_pred_panc))[,4][2], sep = "")
```
We see that there is a statistically significant interaction between T2D and PanC. The odds ratio of 3.84 tells us that there is a positive association between T2D status and PanC risk, and that the odds of a T2D patient developing PANC is 3.84 times greater than in individuals without T2D. 

### Association Between T2D and Pancreatic Cancer With Adjustment by all Covariates

```{r T2D and PanC - all Covs, echo = F}

# regression model of PanC and T2D + all covariates 
t2d_pred_panc_cov_all = glm(C25 ~ T2D + Sex + AgeBaseline + BMI +
                              as.factor(AssesmentCentre),
                            family=binomial("logit"), data = pheno_file)
summary(t2d_pred_panc_cov_all)

# Odds ratios and p-value of T2D 
cat("OR = ", round(exp(coef(summary(t2d_pred_panc_cov_all)))[,1][2], 2), 
    ", P-value = ", coef(summary(t2d_pred_panc_cov_all))[,4][2], sep = "")

# extracting p-values
cat(" P-values: ", sep = "\n")
coef(summary(t2d_pred_panc_cov_all))[,4]

# ORs and CIs
# or_CI_t2d_pred_panc_cov_all <- round(exp(cbind(coef(t2d_pred_panc_cov_all), # confint(t2d_pred_panc_cov_all))),  digits = 2) %>% as.data.frame()


```
From this initial analysis, we see that BMI and UKB Assessment Centre are not statistically significant covariates. In earlier regression analyses and GxE testing, it was also found that UKB Assessment Centre was an insignificant covariate. As such, it will not be included for downstream GxE testing. 

The odds ratio of 2.92 indicates a positive relationship between T2D and PanC, though it is lower when not adjusting for any covariates (as one would expect). The p-value for T2D is statistically significant and very low, indicating a strong relationship. 

### Association of T2D and Pancreatic Cancer Adjusted by Age, Sex and BMI

```{r T2D and PanC - Sex & Age & BMI, echo = F}
# regression model of PanC and T2D with adjustment for sex, age, and BMI 
t2d_pred_panc_out_covs = glm(C25 ~ T2D + Sex + AgeBaseline,
                            family=binomial("logit"), data = pheno_file)
summary(t2d_pred_panc_out_covs)

# extracting p-values
cat(" P-values: ", sep = "\n")
coef(summary(t2d_pred_panc_out_covs))[,4]

# ORs and CIs
or_CI_t2d_pred_panc <- round(exp(cbind(coef(t2d_pred_panc_out_covs), confint(t2d_pred_panc_out_covs))),  digits = 2) %>% as.data.frame()

```
Again, we observe that age and sex are statistically significant covariates in the regression. The OR of 2.97 indicates a positive relationship between PANC risk and T2D. The p-value for T2D is again very low and statistically significant, suggesting a strong relationship. 

### Interaction Between Sex and T2D status on PanC risk?

```{r T2D and Sex Interaction on PanC Risk, echo = F}
# regression model to investigate interaciton between sex and T2D status on PanC risk
t2d_sex_interaction = glm(C25 ~ T2D + Sex + T2D*Sex + AgeBaseline + BMI,
                          family=binomial("logit"), 
                          data = pheno_file)
summary(t2d_sex_interaction)


# extracting p-values
cat("P-values", sep = "\n")
coef(summary(t2d_sex_interaction))[,4]

# nominally significant interaction between T2D and sex on PANC risk 

# checking the ORs for sex in individuals without T2D 
t2d_sex_noT2D = glm(C25 ~ Sex + AgeBaseline + BMI,
                          family=binomial("logit"), 
                          data = pheno_file[pheno_file$T2D == 0, ])
summary(t2d_sex_noT2D)

or_CI_not2d_sex<- round(exp(cbind(coef(t2d_sex_noT2D), confint(t2d_sex_noT2D))),  digits = 2) %>% as.data.frame()

# checking the ORs for sex in individuals with T2D 
# ((sex is not significant here, but age and BMI are)
t2d_sex_T2D = glm(C25 ~ Sex + AgeBaseline + BMI,
                          family=binomial("logit"), 
                          data = pheno_file[pheno_file$T2D == 1, ])
summary(t2d_sex_T2D)

or_CI_t2d_sex<- round(exp(cbind(coef(t2d_sex_T2D), confint(t2d_sex_T2D))),  digits = 2) %>% as.data.frame()
```
There is a nominally statistically significant interaction between sex and T2D status on risk of PanC. This is expected, as it has been found that men are at slightly higher risk of developing PanC compared to women. However, the risk reported in literature only differs by a few percentage points, and so it not too unsurprising that the interaction is only nominally significant. 

```{r Interaction Plot for T2DxAge, echo = F}
# stratifying the analysis to look at the difference between men and women 
# interaction plot of T2D and Sex of PanC outcome
sex_interact_plot = interact_plot(t2d_sex_interaction, pred = T2D, modx = Sex, interval = T, int.width = 0.95, outcome.scale = "response", x.label = "T2D Status", y.label = "PanC Risk")

# formatting the interaction plot 
sex_interact_plot +
  theme_bw() + 
  theme(legend.key.size = unit(0.8, 'cm'), plot.title = element_text(face = "bold")) + 
  ggtitle("Interaction Plot for Sex x T2D Status")
ggsave("Interaction Plot Sex x T2D Status.png")

```


### Interaction Between Age and T2D status on PANC risk? 

```{r T2D and Age Interaction on PanC Risk, echo = F}

# regression model to investigate interaciton between age and T2D status on PanC risk
t2d_age_interaction = glm(C25 ~ T2D + AgeBaseline + T2D*AgeBaseline + Sex,
                          family=binomial("logit"), data = pheno_file)
summary(t2d_age_interaction)

# extracting p-values
cat("P-values", sep = "\n")
coef(summary(t2d_age_interaction))[,4]

# significant interaction between T2D and age on PANC outcome

```
There is indeed a statistically significant interaction between age and T2D status on risk of PANC This makes sense, as PANC is largely seen in patients of older age. 

#### Interaction Plot Age x T2D Status
```{r T2D and Age Interaction on PanC Risk - Interaction Plot, echo = FALSE}

# interaction plot of T2D and Age of PANC outcome
age_interact_plot = interact_plot(t2d_age_interaction, pred = "AgeBaseline", modx = "T2D", interval = T, int.width = 0.95, outcome.scale = "response", x.label = "Age at Baseline", y.label = "PanC Risk")

# formatting the interaction plot 
age_interact_plot +
  theme_bw() + 
  theme(legend.key.size = unit(0.8, 'cm'), plot.title = element_text(face = "bold")) + 
  ggtitle("Interaction Plot for Age x T2D Status")
ggsave("Interaction Plot Age x T2D Status.png")

# other kind of interaction plot 
# ggpredict(t2d_age_interaction, c("AgeBaseline", "T2D")) %>% plot()

```

Age is a strong predictor of PanC risk in individuals without T2D, whilst the prediction is less strong in individuals who already have a T2D diagnosis (Figure 3). However, in both cases, increasing age entails an increased risk of developing PanC. 

### Interaction Between BMI and T2D status on PANC risk? 
```{r T2D and BMI interaction on PanC risk, echo = F}

# regression model to investigate interaciton between BMI and T2D status on PanC risk
t2d_bmi_interaction = glm(C25 ~ T2D + BMI + T2D*BMI + Sex + AgeBaseline,
                          family=binomial("logit"), data = pheno_file)
summary(t2d_bmi_interaction)

# significant interaction between BMI and T2D on PANC outcome 

```
There is indeed a statistically significant interaction between BMI and T2D status on risk of PANC. This makes sense, as BMI has been found to be a risk factor of both T2D and PANC. 

## Exploratory Regression Analyses - RG and PANC
This section contains some simple regression analyses to investigate the relationship between RG and certain covariates on PANC outcome. Note that in all analyses, log(RG) (normalised) is used rather than raw RG values. 

### Association Between RG and Pancreatic Cancer Without Adjustment by Covariates
```{r RG and PanC - no covs, echo = F}

# simple regression analysis between RG level on PanC outcome 
RG_pred_panc_out = glm(PANC.incident.RG ~ RG.control.log,
                       family=binomial("logit"), 
                      data = pheno_file[pheno_file$T2D == 0, ])
summary(RG_pred_panc_out)

# extracting p-values
cat(" P-values: ", sep = "\n")
coef(summary(RG_pred_panc_out))[,4]

# ORs and CIs
or_CI_RG_panc <- round(exp(cbind(coef(RG_pred_panc_out), confint(RG_pred_panc_out))),  digits = 2) %>% as.data.frame()


```
We see that there is a statistically significant relationship between RG levels and PANC. Thse odds ratio of 2.12 tells us that there is a positive association between RG levels ans PANC risk. 

### Association Between RG and Pancreatic Cancer With Adjustment by All Covariates 
```{r RG and PanC - All Covs, echo = F}

# regression analysis to assess PanC risk by RG level and adjustment by all covariates 
RG_pred_panc_cov = glm(PANC.incident.RG ~ RG.control.log + Sex + AgeBaseline +
                         BMI + as.factor(AssesmentCentre) + Fasting.time +
                         Fasting.time^2 + Fasting.time^3,
                       family=binomial("logit"), 
                       data = pheno_file[pheno_file$T2D == 0, ])
summary(RG_pred_panc_cov)

# extracting p-values
cat(" P-values: ", sep = "\n")
coef(summary(RG_pred_panc_cov))[,4]

# ORs and CIs
#or_CI_RG_panc_all_covs <- round(exp(cbind(coef(RG_pred_panc_cov), confint(RG_pred_panc_cov))),  digits = 2) %>% as.data.frame()

```
From this initial analysis, we see that UKB Assessment Centre is not a statistically significant covariate, and neither is fasting time. BMI is only slightly significant. From previous analyses and GxE testing, UKB Assessment Centre was also found to be insignificant. Hence, in the following analyses and GxE testing, UKB Assessment Centre will not be included as a covariate. 

However, we also see that when adjusting for all covariates, the RG variable no longer is a statistically significant predictor of PanC risk. We are therefore unlikely to find any interactions in the downstream analysis (but I'll try anyways!).

### Association of T2D and Pancreatic Cancer Adjusted by Age, Sex, BMI and Fasting Time

```{r RG and PanC - main covs, echo = F}
# regression model to assess effect of RG level and main covariates on PanC risk
RG_pred_panc_out_covs = glm(PANC.incident.RG ~ RG.control.log + Sex + AgeBaseline
                            + BMI + Fasting.time + Fasting.time^2 +
                              Fasting.time^3, family=binomial("logit"), 
                            data = pheno_file[pheno_file$T2D == 0, ])
summary(RG_pred_panc_out_covs)

# extracting p-values
cat(" P-values: ", sep = "\n")
coef(summary(RG_pred_panc_out_covs))[,4]

# ORs and CIs
or_CI_RG_panc_main_covs <- round(exp(cbind(coef(RG_pred_panc_out_covs),
                    confint(RG_pred_panc_out_covs))),  digits = 2) %>% as.data.frame()


```
We observe a slightly positive OR = 1.2, but not a statistically significant p-value. RG is not a predictor of PANC risk when adjusting for sex, age, BMI and fasting time. 

### Interaction Between Sex and RG level on PanC risk?

```{r Interaction Sex and RG on PanC Risk, echo = F}
# regression model to investigate interaciton between sex and RG level on PanC risk
RG_sex_interaction = glm(PANC.incident.RG ~ RG.control.log + Sex + RG.control.log*Sex + 
                           AgeBaseline + BMI + Fasting.time + Fasting.time^2 +
                           Fasting.time^3, family=binomial("logit"), 
                         data =pheno_file[pheno_file$T2D == 0, ])
summary(RG_sex_interaction)

# no interaction between RG and sex on PanC risk 

```
There is no statistically significant interaction between sex and RG level on risk of PANC. 

### Interaction Between Age and RG level on PanC risk? (no)
```{r Interaction Age and RG on PanC Risk, echo = F}
# regression model to investigate interaciton between age and log(RG) on PanC risk
RG_age_interaction = glm(PANC.incident.RG ~ RG.control.log + AgeBaseline +
                           RG.control.log*AgeBaseline + Sex + BMI + Fasting.time +
                           Fasting.time^2 + Fasting.time^3,
                         family=binomial("logit"), 
                         data = pheno_file[pheno_file$T2D == 0, ])
summary(RG_age_interaction)

# no interaction between age and RG level on PanC risk 

```
There is no statistically significant interaction between age and RG level on PanC risk. 

### Interaction Between BMI and RG level on PanC risk? 
```{r Interaction BMI and RG on PanC Risk, echo = F}
RG_bmi_interaction = glm(PANC.incident.RG ~ RG.control.log + BMI + RG.control.log*BMI +
                           Sex + AgeBaseline + Fasting.time + Fasting.time^2 +
                           Fasting.time^3, family=binomial("logit"), 
                         data = pheno_file[pheno_file$T2D == 0, ])
summary(RG_bmi_interaction)

# no significant interaction between BMI and T2D on PanC risk

```
There is no statistically significant interaction between BMI and RG level on PanC risk. 


## GxE Testing with T2D PRS 
This section contains all GxE analyses using T2D PRS as the genetic variable. 

```{r Setup T2D PRS, echo = F}
# loading T2D PRS and merging with phenotype file 
t2d_prs = fread("ukbb_t2d_weighted_score.profile") %>% mutate(score_norm = scale(SCORESUM))
t2d_prs_with_pheno = merge(t2d_prs, pheno_file, by.x = "IID", by.y = "ID_1")
```

### GxE Testing of T2D with T2D PRS 

```{r GxE T2D with T2D PRS, echo = F}
# testing without T2D status and without GxE interaction term, only main covs
t2d_prs_panc = glm(C25 ~ score_norm + Sex + AgeBaseline, family=binomial("logit"), 
                   data = t2d_prs_with_pheno)
summary(t2d_prs_panc)

# testing with T2D and with GxE interaction term + all covs 
t2d_prs_panc_t2d_gxe = glm(C25 ~ score_norm + T2D + T2D*score_norm + Sex + BMI +
                             AgeBaseline + Array + PC1 + PC2 + PC3 + PC4 + PC5
                           + PC6,
                       family=binomial("logit"), data = t2d_prs_with_pheno)
summary(t2d_prs_panc_t2d_gxe)

# extracting p values
cat("P-values", sep = "\n")
coef(summary(t2d_prs_panc_t2d_gxe))[,4]

# interaction plot of GxE model
t2d_prs_t2d_interact_plot = interact_plot(t2d_prs_panc_t2d_gxe, pred = score_norm, modx = T2D, interval = T, int.width = 0.95, outcome.scale = "response", x.label = "T2D PRS", y.label = "PanC Risk", y.lim = c(0,1), yaxt = "n")

# formatting the interaction plot 
t2d_prs_t2d_interact_plot +
  theme_bw() + 
  theme(legend.key.size = unit(0.8, 'cm'), plot.title = element_text(face = "bold", size = 12)) + 
  ggtitle("Interaction Plot of T2D PRS x T2D Status - Original Definition of PanC Incident") + 
  scale_x_continuous(limits = c(-5.5, 4.5)) 
ggsave("T2D PRS x T2D Status Interaction Plot - Original PanC Incident.png")

# extracting ORs and CIs for the interaction model 
or_CI_t2d_PRS_t2d <- round(exp(cbind(coef(t2d_prs_panc_t2d_gxe), confint(t2d_prs_panc_t2d_gxe))), digits = 2) %>% as.data.frame()

or_CI_t2d_PRS_t2d <- or_CI_t2d_PRS_t2d %>% 
  mutate(variable=rownames(or_CI_t2d_PRS_t2d))
  
or_CI_t2d_PRS_t2d <- rename(or_CI_t2d_PRS_t2d, 
                            c(AOR = V1,
                        lower_bound = `2.5 %`,
                        upper_bound = `97.5 %`))

# reorder variables in the OR df 
col_order <- c("variable", "AOR", "lower_bound", "upper_bound")
or_CI_t2d_PRS_t2d <- or_CI_t2d_PRS_t2d[, col_order] 

# plotting ORs and CIs - forest plot
plot_logit_model <- or_CI_t2d_PRS_t2d[-c(1, 7, 8, 9, 10, 11, 12, 13),] %>%  #remove the rows we don't wanna plot
  ggplot(aes(x = reorder(variable, AOR), y = AOR)) +
  geom_point(shape = 16,
             size  = 3.5, width = 0.1,
             position = "dodge", color="black") + 
  geom_errorbar(aes(ymin  = lower_bound,
                    ymax  = upper_bound),
                width = 0.2,
                size  = 0.7,
                position = "dodge", color="turquoise4") +
  theme_bw() + 
  theme(plot.title = element_text(face = "bold")) +
  xlab("") + ylab("Adjusted ORs with 95% CI") +
  ggtitle("GxE Model of T2D PRS and T2D Status on PanC Outcome") + 
  coord_flip(ylim = c(0, 3.5)) + 
  geom_hline(yintercept = 1, color = "#6F8FAF", size = 1, linetype = "dashed") +
  theme(axis.title = element_text(size = 10)) + 
  theme(axis.text = element_text(size = 10)) + 
  scale_x_discrete(breaks = c("T2D", "Sex", "AgeBaseline", "BMI", "score_norm", "score_norm:T2D"), 
                   labels = c("T2D Status", "Sex", "Baseline Age", "BMI", "T2D PRS", "T2D Status*T2D PRS")) 
ggsave("Forest Plot T2D PRS x T2D Status.png")
plot_logit_model

```

There is a statistically significant interaction between T2D and T2D PRS on PanC outcome. However, when we plot the predictive value of the PRS and T2D status for PanC outcome, we see that the T2D PRS is not predictive of who will develop PANC amongst individuals without T2D. 

However, in T2D patients, the lower genetic predisposition to T2D, the higher the risk of PANC. This implies that T2D is often just a symptom of PANC, even if it has been diagnosed before PANC. To further confirm this hypothesis, we create a new PanC after T2D definition, and again test for GxE interaction betwene T2D PRS and T2D status on risk of PanC. 

### GxE Testing with T2D PRS with a New Definiton of PANC Incidcent Cases 

```{r GxE Testing T2D PRS With New Definition of PanC Incident, echo = F}

# defining new cases of PANC + T2D 
# (excluding cases where T2D was diagnosed more than one year after PANC)
pheno_file$C25_exclude = ifelse((pheno_file$Date.C25 < pheno_file$Date.E11 + 365) |
                                (pheno_file$Date.C25.hosp < pheno_file$Date.E11 + 365), T, F)
sum(pheno_file$C25_exclude, na.rm = T)

# setting the NA values to false
pheno_file$C25_exclude[is.na(pheno_file$C25_exclude)] = F

# merging phenotype file with T2D PRS, only including the PanC + T2D cases where T2D was diagnosed prior to or less than one year after PanC
t2d_prs_with_pheno_C25_exclude = merge(t2d_prs, pheno_file[!pheno_file$C25_exclude, ], 
                           by.x = "IID", by.y = "ID_1")

# testing without T2D status and without GxE interaction term, main covs
t2d_prs_panc_new = glm(C25 ~ score_norm  + Sex + AgeBaseline, family=binomial("logit"), 
                       data = t2d_prs_with_pheno_C25_exclude)
summary(t2d_prs_panc_new)

# testing with T2D status and with GxE interaction term, all covs
t2d_prs_panc_gxe_new = glm(C25 ~ score_norm + T2D + T2D*score_norm + Sex + BMI
                            + AgeBaseline + Array + PC1 + PC2 + PC3 + PC4 + PC5
                           + PC6,
                           family=binomial("logit"), data = t2d_prs_with_pheno_C25_exclude)
summary(t2d_prs_panc_gxe_new)

# extracting p values
cat("P-values", sep = "\n")
coef(summary(t2d_prs_panc_gxe_new))[,4]

# interaction plot of T2D and T2D PRS as predictor 
t2d_prs_t2d_interact_plot_new = interact_plot(t2d_prs_panc_gxe_new, pred = score_norm, modx = T2D, interval = T, int.width = 0.95, outcome.scale = "response", x.label = "T2D PRS", y.label = "PanC Risk") 

# formatting the interaction plot 
t2d_prs_t2d_interact_plot_new +
  theme_bw() + 
  theme(legend.key.size = unit(0.8, 'cm'), plot.title = element_text(face = "bold", size = 12)) + 
  ggtitle("Interaction Plot of T2D PRS x T2D Status - New Definition of PanC Incident") + 
  scale_x_continuous(limits = c(-5.5, 4.5))
ggsave("T2D PRS x T2D Status Interaction Plot - New PanC Incident.png")

# extracting ORs and CIs 
or_CI_t2d_PRS_t2d_exclude <- round(exp(cbind(coef(t2d_prs_panc_gxe_new), confint(t2d_prs_panc_gxe_new))), digits = 2) %>% as.data.frame()

or_CI_t2d_PRS_t2d_exclude <- or_CI_t2d_PRS_t2d_exclude %>% 
  mutate(variable=rownames(or_CI_t2d_PRS_t2d_exclude)) 
  
or_CI_t2d_PRS_t2d_exclude <- rename(or_CI_t2d_PRS_t2d_exclude, 
                            c(AOR = V1,
                        lower_bound = `2.5 %`,
                        upper_bound = `97.5 %`))

# reorder variables in the OR df 
col_order <- c("variable", "AOR", "lower_bound", "upper_bound")
or_CI_t2d_PRS_t2d_exclude <- or_CI_t2d_PRS_t2d_exclude[, col_order] 

plot_logit_model <- or_CI_t2d_PRS_t2d_exclude[-c(1, 7, 8, 9, 10, 11, 12, 13),] %>%  #remove the rows we don't wanna plot
  ggplot(aes(x = reorder(variable, AOR), y = AOR)) +
  geom_point(shape = 16,
             size  = 3.5, width = 0.1,
             position = "dodge", color="black") + 
  geom_errorbar(aes(ymin  = lower_bound,
                    ymax  = upper_bound),
                width = 0.2,
                size  = 0.7,
                position = "dodge", color="turquoise4") +
  theme_bw() + 
  theme(plot.title = element_text(face = "bold")) +
  xlab("") + ylab("Adjusted ORs with 95% CI") +
  ggtitle("GxE Model of T2D PRS and T2D Status on PanC Outcome - New PanC Definition") + 
  coord_flip(ylim = c(0, 3.5)) + 
  geom_hline(yintercept = 1, color = "#6F8FAF", size = 1, linetype = "dashed") +
  theme(axis.title = element_text(size = 10)) + 
  theme(axis.text = element_text(size = 10)) + 
  scale_x_discrete(breaks = c("T2D", "Sex", "AgeBaseline", "BMI", "score_norm", "score_norm:T2D"),
                   labels = c("T2D Status", "Sex", "Baseline Age", "BMI", "T2D PRS", "T2D
                              Status*T2D PRS")) 
ggsave("Forest Plot T2D PRS x T2D Status.png")
plot_logit_model



```

Indeed, we see that in patients with newly onset T2D after a PanC diagnosis (less than 365 days after PANC diagnosis) or with a T2D diagnosis that occured prior to PANC diagnosis, T2D PRS is a predictor of higher PanC risk. This confirms the above hypothesis, that T2D is an early sign of pancreatic cancer. 

In T2D patients, the higher the T2D PRS, the more likely to develop PanC. 

### GxE Testing of RG with T2D PRS

```{r GxE Testing of RG with T2D PRS, echo = F}

# testing without RG and without GxE interaction term, only main covs 
t2d_prs_panc_RG = glm(PANC.incident.RG ~ score_norm + Sex + AgeBaseline +
                        Fasting.time + Fasting.time^2 + Fasting.time^3,
                   family=binomial("logit"), 
                   data = t2d_prs_with_pheno)
summary(t2d_prs_panc_RG)

# testing with RG and with all covs
t2d_prs_panc_RG_covs = glm(PANC.incident.RG ~ score_norm + RG.control.log + Sex + BMI + 
                             AgeBaseline + Fasting.time + Fasting.time^2 + Fasting.time^3
                           + Array + PC1 + PC2 + PC3 + PC4 + PC5 + PC6,
                   family=binomial("logit"), 
                   data = t2d_prs_with_pheno)
summary(t2d_prs_panc_RG_covs)

# GxE testing with RG and with GxE interaction term, all covs 
t2d_prs_panc_RG_gxe = glm(PANC.incident.RG ~ score_norm + RG.control.log +
                            score_norm*RG.control.log* + Sex + AgeBaseline + BMI
                          + Fasting.time +
                            Fasting.time^2 + Fasting.time^3 + 
                            Array + PC1 + PC2 + PC3 + PC4 + PC5 + PC6,
                       family=binomial("logit"), data = t2d_prs_with_pheno)
summary(t2d_prs_panc_RG_gxe)

# extracting p values
cat("P-values", sep = "\n")
coef(summary(t2d_prs_panc_RG_gxe))[,4]

# interaction plot of GxE model
interact_plot(t2d_prs_panc_RG_gxe, pred = score_norm, modx = RG.control.log, interval = T, int.width = 0.95, outcome.scale = "response", x.label = "T2D PRS", y.label = "PanC Risk")

# forest plots
# extracting ORs and CIs for the interaction model 
or_CI_t2d_PRS_RG <- round(exp(cbind(coef(t2d_prs_panc_RG_gxe), confint(t2d_prs_panc_RG_gxe))), digits = 2) %>% as.data.frame()

or_CI_t2d_PRS_RG <- or_CI_t2d_PRS_RG %>% 
  mutate(variable=rownames(or_CI_t2d_PRS_RG)) # extract the variables from rownames

or_CI_t2d_PRS_RG <- rename(or_CI_t2d_PRS_RG, 
                             c(AOR = V1,
                        lower_bound = `2.5 %`,
                        upper_bound = `97.5 %`))

# reorder variables in the df
col_order <- c("variable", "AOR", "lower_bound", "upper_bound")
or_CI_t2d_PRS_RG <- or_CI_t2d_PRS_RG[, col_order] #reorder variables in the data frame

plot_logit_model_RG_T2D <- or_CI_t2d_PRS_RG[-c(1, 8, 9, 10, 11, 12, 13, 14),] %>%  #remove the rows we don't wanna plot
  ggplot(aes(x = reorder(variable, AOR), y = AOR)) +
  geom_point(shape = 16,
             size  = 3.5, width = 0.1,
             position = "dodge", color="black") + 
  geom_errorbar(aes(ymin  = lower_bound,
                    ymax  = upper_bound),
                width = 0.2,
                size  = 0.7,
                position = "dodge", color="turquoise4") +
  theme_bw() + 
  theme(plot.title = element_text(face = "bold")) +
  xlab("") + ylab("Adjusted ORs with 95% CI") +
  ggtitle("GxE Model of T2D PRS and RG on PanC Outcome") + 
  coord_flip(ylim = c(0, 3.5)) + 
  geom_hline(yintercept = 1, color = "#6F8FAF", size = 1, linetype = "dashed") +
  theme(axis.title = element_text(size = 10)) + 
  theme(axis.text = element_text(size = 10)) + 
  scale_x_discrete(breaks = c("RG.control.log", "Sex", "AgeBaseline", "BMI", "score_norm",
                              "score_norm:RG.control.log", "Fasting.time"), 
                   labels = c("Random Glucose", "Sex", "Baseline Age", "BMI", "T2D PRS", 
                              "T2D PRS*RG", "Fasting time")) 
ggsave("Forest Plot T2D PRS x Random Glucose.png")
plot_logit_model_RG_T2D

```

We see that for RG values, the lower the genetic predisposition to T2D (lower T2D PRS), the higher the chance of developing PanC. However, the interactions and associations are not significant. RG is not a significant predictor in this relationship. Additionally, the 95% CIs for T2D PRS, RG, and T2D PRS*RG are very wide, suggesting a high degree of uncertainty. 

## GxE Testing with PanC PRS 
This section contains all GxE analyses using PanC PRS as the genetic variable. 

```{r GxE PanC PRS Setup}
# loading PanC PRS and merging with phenotype file 
panc_prs = fread("ukbb_panc_snps_panc_betas_score.profile") %>% mutate(score_norm = scale(SCORESUM))
panc_prs_with_pheno = merge(panc_prs, pheno_file, by.x = "IID", by.y = "ID_1")

```

### GxE Testing of T2D with PanC PRS 

```{r GxE Testing of T2D with PanC PRS, echo = FALSE}

# testing without T2D status and without interaction term, main covs
panc_prs_panc = glm(C25 ~ score_norm + Sex + AgeBaseline + BMI,
                    family=binomial("logit"), 
                    data = panc_prs_with_pheno)
summary(panc_prs_panc)

# GxE testing with T2D status and interaction term, all covs 
panc_prs_t2d_panc_gxe = glm(C25 ~ score_norm + T2D + T2D*score_norm + Sex + AgeBaseline
                            + BMI + Array + PC1 + PC2 + PC3 + PC4 + PC5 + PC6,
                            family=binomial("logit"), data = panc_prs_with_pheno)
summary(panc_prs_t2d_panc_gxe)

# extracting p values
cat("P-values", sep = "\n")
coef(summary(panc_prs_t2d_panc_gxe))[,4]

# interaction plot 
panc_prs_t2d_interact_plot = interact_plot(panc_prs_t2d_panc_gxe, pred = score_norm, modx = T2D, interval = T, int.width = 0.95, outcome.scale = "response", x.label = "PanC PRS", y.label = "PanC Risk", rescale.axis = F) 

# formatting the interaction plot 
panc_prs_t2d_interact_plot +
  theme_bw() + 
  theme(legend.key.size = unit(0.8, 'cm'), plot.title = element_text(
    face = "bold", size = 12)) + 
  ggtitle("Interaction Plot of PanC PRS x T2D Status") + 
  scale_x_continuous(limits = c(-3, 4.5)) 
ggsave("PanC PRS x T2D Status Interaction Plot.png")

# forest plot 
# extracting ORs and CIs for the interaction model 
or_CI_panc_PRS_t2d <- round(exp(cbind(coef(panc_prs_t2d_panc_gxe), confint(panc_prs_t2d_panc_gxe))), digits = 2) %>% as.data.frame()

or_CI_panc_PRS_t2d <- or_CI_panc_PRS_t2d %>% 
  mutate(variable=rownames(or_CI_panc_PRS_t2d)) # extract the variables from rownames

or_CI_panc_PRS_t2d <- rename(or_CI_panc_PRS_t2d, 
                             c(AOR = V1,
                        lower_bound = `2.5 %`,
                        upper_bound = `97.5 %`))

# reorder variables in the df
col_order <- c("variable", "AOR", "lower_bound", "upper_bound")
or_CI_panc_PRS_t2d <- or_CI_panc_PRS_t2d[, col_order] #reorder variables in the data frame

# plotting ORs and CIs
plot_panc_prs_T2D_gxe <- or_CI_panc_PRS_t2d[-c(1, 7, 8, 9, 10, 11, 12, 13),] %>%  #removing the variables we don't need
  ggplot(aes(x = reorder(variable, AOR), y = AOR)) +
  geom_point(shape = 16,
             size  = 3.5, width = 0.1,
             position = "dodge", color="black") + 
  geom_errorbar(aes(ymin  = lower_bound,
                    ymax  = upper_bound),
                width = 0.2,
                size  = 0.7,
                position = "dodge", color="turquoise4") +
  theme_bw() + 
  theme(plot.title = element_text(face = "bold")) +
  xlab("") + ylab("Adjusted ORs with 95% CI") +
  ggtitle("GxE Model of PanC PRS and T2D Status on PanC Outcome") + 
  coord_flip(ylim = c(0, 3.5)) + 
  geom_hline(yintercept = 1, color = "#6F8FAF", size = 1, linetype = "dashed") +
  theme(axis.title = element_text(size = 10)) + 
  theme(axis.text = element_text(size = 10)) + 
  scale_x_discrete(breaks = c("T2D", "Sex", "AgeBaseline", "score_norm", "score_norm:T2D", "BMI"),
                   labels = c("T2D Status", "Sex", "Baseline Age", "PanC PRS", "T2D Status*PanC
                              PRS", "BMI")) 
ggsave("Forest Plot PanC PRS x T2D Status.png")
plot_panc_prs_T2D_gxe
```

The results are as expected. PanC PRS is a highly significant predictor of PanC risk. The higher the PanC PRS, the more more likely to develop PanC. We see that in T2D patients, the PanC PRS is an even better predictor of PanC outcome than in individuals without T2D. 

### GxE Testing of RG with PanC PRS 
```{r GxE PanC PRS x RG, echo = FALSE}

# testing without RG and without GxE interaction term, main covs
panc_prs_panc_RG = glm(PANC.incident.RG ~ score_norm + Sex + AgeBaseline + Fasting.time + 
                         Fasting.time^2 + Fasting.time^3,
                       family=binomial("logit"), 
                       data = panc_prs_with_pheno)
summary(panc_prs_panc_RG)

# GxE testing with RG and with GxE interaction term, all covs
panc_prs_panc_RG_gxe = glm(PANC.incident.RG ~ score_norm + RG.control.log +
                             score_norm*RG.control.log + Sex + AgeBaseline + BMI + 
                             Fasting.time + Fasting.time^2 + Fasting.time^3 + Array + 
                             PC1 + PC2 + PC3 + PC4 + PC5 + PC6,
                       family=binomial("logit"), 
                       data = panc_prs_with_pheno)
summary(panc_prs_panc_RG_gxe)

# extracting p values
cat("P-values", sep = "\n")
coef(summary(panc_prs_panc_RG_gxe))[,4]

# interaction plot 
interact_plot(panc_prs_panc_RG_gxe, pred = score_norm, modx = RG.control.log, interval = T, int.width = 0.95, outcome.scale = "response", x.label = "PanC PRS Score", y.label = "PanC Risk")

# forest plots
# extracting ORs and CIs for the interaction model 
or_CI_panc_PRS_RG <- round(exp(cbind(coef(panc_prs_panc_RG_gxe), confint(panc_prs_panc_RG_gxe))), digits = 2) %>% as.data.frame()

or_CI_panc_PRS_RG <- or_CI_panc_PRS_RG %>% 
  mutate(variable=rownames(or_CI_panc_PRS_RG)) # extract the variables from rownames

or_CI_panc_PRS_RG <- rename(or_CI_panc_PRS_RG, 
                             c(AOR = V1,
                        lower_bound = `2.5 %`,
                        upper_bound = `97.5 %`))

# reorder variables in df 
col_order <- c("variable", "AOR", "lower_bound", "upper_bound")
or_CI_panc_PRS_RG <- or_CI_panc_PRS_RG[, col_order] #reorder variables in the data frame

# plotting ORs and CIs
plot_panc_prs_RG_gxe <- or_CI_panc_PRS_RG[-c(1, 8, 9, 10, 11, 12, 13, 14),] %>%  #remove the columns we don't need 
  ggplot(aes(x = reorder(variable, AOR), y = AOR)) +
  geom_point(shape = 16,
             size  = 3.5, width = 0.1,
             position = "dodge", color="black") + 
  geom_errorbar(aes(ymin  = lower_bound,
                    ymax  = upper_bound),
                width = 0.2,
                size  = 0.7,
                position = "dodge", color="turquoise4") +
  theme_bw() + 
  theme(plot.title = element_text(face = "bold")) +
  xlab("") + ylab("Adjusted ORs with 95% CI") +
  ggtitle("GxE Model of PanC PRS and RG on PanC Outcome") + 
  coord_flip(ylim = c(0, 3.5)) + 
  geom_hline(yintercept = 1, color = "#6F8FAF", size = 1, linetype = "dashed") +
  theme(axis.title = element_text(size = 10)) + 
  theme(axis.text = element_text(size = 10)) + 
  scale_x_discrete(breaks = c("score_norm", "RG.control.log", "Sex", "AgeBaseline", "score_norm:RG.control", "BMI", "Fasting.time"), 
                   labels = c("PanC PRS", "Random Glucose", "Sex", "Baseline Age", "RG*PanC PRS", "BMI", "Fasting Time"))
ggsave("Forest Plot PanC PRS x Random Glucose.png")
plot_panc_prs_RG_gxe
```

Again, as expected, PanC PRS is a predictor of PanC risk. However, RG levels are not significant in this relationship. 

## Extra - BMI as a possible third environment to test? 
Towards the end of the project, I wondered whether BMI was a possible third environment I could use to test for GxE interaction with T2D and PanC PRS for PanC outcome. Although it is not a glycaemic trait, BMI is known to be a risk-factor of both T2D and PanC, and is likely to be involved in their association. However, time did not allow me to further explore this relationship - but it is certainly a possibility for future work! 

```{r BMI as third environment?, echo = F}
# I am wondering whether BMI possibly is an "environment" I can use to test for interaction 

# just checking if BMI is a possible environment to test? Maybe? 
bmi_pred_panc = glm(C25 ~ BMI, family=binomial("logit"), data = pheno_file)
summary(bmi_pred_panc)

# check with main covariates
bmi_pred_panc_cov = glm(C25 ~ BMI + AgeBaseline + Sex, family=binomial("logit"), 
                        data = pheno_file)
summary(bmi_pred_panc_cov)

# the above looks good...potential for future exploration! 

```

