---
title: "Exploratory Data Analysis - PanC and T2D"
author: "Edwina Rossi"
output:
  pdf_document:
    fig_width: 6
    fig_height: 4
    extra_dependencies: ["float", "flafter"]
  html_document: default
 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE, fig.pos = "!H", out.extra = "", fig.show="hold", out.width="50%")

```

# Exploratory Data Analysis

Exploratory Data Analysis (EDA) is an important step when working with large scale data sets, such as the epidemiological data set used in this project. The aim of the EDA is to investigate the data and summarise its main characteristics through the means of plots and summary statistics. The main results from the EDA can be found in the final report "Results" section and in "Appendix D - Exploratory Data Analysis"

```{r import epidemiological data, include=FALSE}
# importing relevant libraries
library(dplyr)
library(data.table)
library(ggplot2)
library(tidyr)
library(scales)
require(ISLR) #for logistic regression 
require(patchwork)
library(tinytex)

# import the epidemiological data
epi_data <- fread("~/Desktop/Masters Project/GxE Testing PanC/Pheno_file_RG_T2D_PANC.txt")

# adding column which represents T2D + PANC cases
epi_data$T2D.PANC = ifelse(epi_data$T2D == 1 & epi_data$C25 == 1, 1, 0)
sum(epi_data$T2D.PANC)

# adding column which represents T2D + PANC cases, where T2D was diagnosed > 365 days after PANC
epi_data$T2D.PANC.1year = ifelse((epi_data$Date.C25 < epi_data$Date.E11 + 365) |
                      (epi_data$Date.C25.hosp < epi_data$Date.E11 + 365), 1, 0)
sum(epi_data$T2D.PANC.1year, na.rm = T)

# adding column which contains RG measurements of all participants without T2D diagnosis 
epi_data$RG.control = ifelse((!is.na(epi_data$Random.glucose) & (epi_data$T2D == 0)),
                             epi_data$Random.glucose, NA)

# counting number of RG measurements without T2D diagnosis
no_RG_measurements_no_T2D = sum(!is.na(epi_data$RG.control))

# adding column which represents all PANC patients with an RG measurement and without a T2D diagnosis (where RG measurement was taken before the PanC diagnosis)
# Make new column containing PANC Incident's RG value
epi_data$PANC.incident.RG.measurements = ifelse((epi_data$T2D == 0) & 
          (epi_data$BaselineDate < epi_data$Date.C25 | 
           epi_data$BaselineDate < epi_data$Date.C25.hosp), 
          epi_data$Random.glucose, NA)

# Binary variable for PANC incident RG 
epi_data$PANC.incident.RG = ifelse(!is.na(epi_data$PANC.incident.RG.measurements), 1, 0)

# counting number of PANC Incident RG cases
PANC_incident_RG = sum(epi_data$PANC.incident.RG == 1, na.rm = T)
      
```

## Counting number of Cases and Controls

```{r case and control count}

# Note that T2D = E11, and PanC = C25

## T2D and PANC
#table of PanC and T2D cases and controls
with(epi_data, addmargins(table(T2D, C25)))

```

## Study Participant Information Plots

### Sex

The distribution of sex (as determined by genotyping) in the epidemiological data is as follows:

```{r genetic sex, echo=FALSE}

# horizontal bar plot of sex
gs <- ggplot(data = epi_data, aes(x = Sex)) +
  geom_bar(width = 0.9, fill = "#B6D0E2") +
  labs(x = "", y = "Number of participants") + 
  #coord_flip() + 
  scale_y_continuous(labels = scales::comma_format(big.mark = ",", decimal.mark = ".")) +
  scale_x_continuous(breaks = seq(0,1),labels = c("Female", "Male")) +
  ggtitle("Sex of Participants") +
  theme_bw() + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
gs
ggsave("Sex Participants.jpg")

```

### Age Distribution

The participants of the UK Biobank study were all aged 40-69 at time of recruitment. The following bar chart illustrates the distribution of participants across age groups 40-49, 50-59 and 60-69 at time of recruitment, and also stratified by sex.

```{r Age at Recruitment, echo=FALSE}
# splitting the age variable into age groups
epi_data$agegrp <- cut(epi_data$AgeBaseline, breaks = c(40, 49, 59, 69, 75), labels = c("40-49", "50-59", "60-69", "70+"), right = T)

ggplot(data = subset(epi_data, !is.na(agegrp)), aes(x = agegrp, na.rm = T)) + 
  geom_bar(fill = "#6F8FAF") + 
  ylab("Number of participants") +
  xlab("Age") +
  scale_y_continuous(expand = expansion(mult = c(0,0.1))) +
  ggtitle("Age Distribution in Study Population") +
  theme(axis.title.y = element_text(vjust = +5)) +
  theme_bw() + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)), axis.title.x = element_text(margin =
          margin(t = 5)))
ggsave("Age Distribution.jpg")


```

### UK Biobank Assessment Centre

The participants in the UK Biobank were assessed at 22 different locations across the UK. The following histogram shows the distribution of participants across the 22 assessment centres:

```{r UK Biobank Assessment Centre, echo=FALSE}

# horizontal histogram of UK BB assessment centre
gs <- ggplot(data = epi_data, aes(x = AssesmentCentre)) +
  geom_bar(binwidth = 0.6, fill = "#088F8F", position=position_dodge(0.8), width=0.8) +
  labs(x = "", y = "Participant Count") + 
  coord_flip() + 
  scale_y_continuous(labels = scales::comma_format(big.mark = ",", decimal.mark = ".")) +
  scale_x_continuous(breaks = c(11001, 11002, 11003, 11004, 11005, 11006, 11007, 11008, 11009,
                                11010, 11011, 11012, 11013, 11014, 11016, 11017, 11018, 11020, 
                                11021, 11022, 11023), 
                     labels = c("Manchester", "Oxford", "Cardiff", "Glasgow", "Edinburgh", "Stoke",                                  "Reading", "Bury", "Newcastle", "Leeds", "Bristol", "Barts",
                                "Nottingham", "Sheffield", "Stoke", "Middelsborough", "Hounslow",
                                "Croydon", "Birmingham", "Swansea", "Wrexham"), 
                     limits = c(11000,11025)) +
  ggtitle("UK Biobank Assessment Centre") +
  theme_bw()
gs
ggsave("UKB Assessment Centre.jpg")


```

## Disease Prevalence

The below plot shows the prevalence of T2D, PanC and T2D + PanC diagnoses within the UK BB study population. 

```{r Disease Prevalence in Study Population, echo=FALSE}

# counting number of cases in each column and placing these into a vector to be plotted
count_disease = c(sum(epi_data$T2D == 1), sum(epi_data$C25 == 1), 
                  sum(epi_data$T2D.PANC == 1, na.rm = T))
names(count_disease) = c("T2D", "PanC", "T2D + PanC")

# combine these two into a data frame
df_disease = data.table(disease = names(count_disease), count = count_disease)

# creating barplot of disease prevalence, with values added
ggplot(data = df_disease, aes(x = disease, y = count)) +
  geom_bar(stat = "identity", fill = "#088F8F", width = 0.5) +
  geom_text(aes(label=count), position=position_dodge(width = 1), vjust=-0.5, size = 3) +
  ylab("Prevalence") +
  xlab("") +
  scale_y_continuous(expand = expansion(mult = c(0,0.1))) +
  ggtitle("Disease Prevalence in Study Population") +
  theme(axis.title.y = element_text(vjust = +5)) +
  theme_bw() + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("Disease Prevalence new.jpg")

```

### Percentage-Wise Disease Prevalence

The below plots show disease prevalence of PanC, T2D and T2D + PanC in percentage terms.

```{r Percentage-Wise Disease Prevalence, echo=FALSE}
# getting disease prevalence in percentage terms 
df_disease$percentages = df_disease$count/nrow(epi_data)

# plotting
ggplot(df_disease, aes(x = disease, y = percentages, label = scales::percent(percentages)), na.rm = T) +
  geom_bar(stat = "identity", fill = "#6082B6", width = 0.5, position = "dodge") +
  ylab("Percentage of population") +
  xlab("") +
  ggtitle("Percentage-Wise Disease Prevalence in Study Population") +
  scale_y_continuous(labels = scales::percent_format(), expand = expansion(mult = c(0,0.1))) +
  geom_text(position = position_dodge(width = .9), vjust = -0.9, size = 3) +
  theme_bw() + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("Percentage-Wise Disease Prevalence.jpg")

# fix percentage scale (make it ~2% bigger)

```

### Are people with PANC more likely to have a T2D diagnosis, or the other way around?

To understand whether T2D patients are more likely to develop PanC, or whether PanC patients are more likely to develop T2D, we may compare the prevalence of the two diseases in different groups.

```{r}

#table of PanC and T2D cases and controls
with(epi_data, addmargins(table(T2D, C25)))

#table of PanC and T2D cases and controls
with(epi_data, addmargins(table(C25, T2D)))

# Percentage table of PanC and T2D cases and controls
with(epi_data, prop.table(table(T2D, C25), margin = 1)*100)

with(epi_data, prop.table(table(C25, T2D), margin = 1)*100)

#Chi-squared test
with(epi_data, chisq.test(T2D, C25))
```

We observe that participants with a T2D diagnosis are \~3 times as likely to also have a PanC diagnosis, compared to participants with out T2D. Equally, we also observe that participants with a PanC diagnosis are \~3 times as likely to also have a T2D diagnosis, compared to participants without PanC

The Chi-squared test between T2D and PANC gives a highly significant p-value, indicating that there indeed is a statistically significant relationship between T2D and PANC.

### Comparison to Disease Prevalence in UK Population

To understand whether the disease prevalence in the study population is representative of the disease prevalence in the actual UK population, we may compare the findings herein to data on PanC and T2D prevalence in the UK population.

According to data from Cancer Research UK, 1 in 53 UK males and 1 in 57 UK females will be diagnosed with pancreatic cancer during their lifetime. On average, this means that roughly 1 in every 55 person is likely to develop pancreatic cancer. If we take the UK population to be 67.22 million (per 2020), this equates to about 1.82%. This figure is almost 5 times higher than in this project's study population. However, it should be noted that 47% of all new pancreatic cancer cases occur in people aged 85 to 89 (Cancer Research UK), and that this project's study population has a lower average age (at recruitment).

With regards to the prevalence of T2D, data from diabetes.org indicate that 1 in 10 over 40s in the UK currently live with diabetes. Knowing that all subjects in the study population are also over 40 years of age, we can compare disease prevalence in this project's study population to the prevalence in the actual UK population in the over 40s. We note that just over 8% of this project's study population have a diabetes diagnosis, compared to the 10% of people living with diabetes in the UK. Though these figures are relatively similar, there might be cases of undiagnosed diabetes in this project's population, explaining why the prevalence of T2D is a bit lower than in the average population.

## Random Glucose

Random Glucose (RG) is a marker of blood glucose values, taken at a random time-point without the patient needing to fast before the blood sample is drawn. It is a common diagnostic marker of diabetes, with values \> 11.1 mmol/L being indicative of T2D. The below plots shows the distribution of RG levels in the study population.

```{r Random Glucose Distribution, echo = FALSE}
# should I quantify the levels into different ranges of RG levels (i.e. based on the diagnostic ranges?)

# plotting RG levels in bins of 1.0 increase, no stratificaiton
ggplot(epi_data[!is.na(epi_data$Sex), ], aes(x = Random.glucose)) +
  geom_histogram(binwidth = 0.1, color="darkblue", fill="lightblue", linetype = "solid") + 
  scale_x_continuous(limits = c(2, 8)) +
  xlab("Random Glucose (mmol/L)") +
  ylab("Participant count") + 
  ggtitle("RG Distribution in Study Population") + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("RG Distribution.jpg")

# plotting normalised RG levels (log 10)
ggplot(epi_data[!is.na(epi_data$Sex), ], aes(x = Random.glucose.log)) +
  geom_histogram(binwidth = 0.01, color="darkblue", fill="lightblue", linetype = "solid") + 
  scale_x_continuous(limits = c(1, 2.3)) +
  xlab("log10(Random Glucose)") + #check the scale here 
  ylab("Participant count") + 
  ggtitle("RG Distribution in Study Population") + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("Normalised RG Distribution log10.jpg")

# labels for facet_grid 
labels = c("0" = "Female", "1" = "Male")

# plotting RG levels in bins of 0.1 increase, separated by sex 
ggplot(epi_data[!is.na(epi_data$Sex), ], aes(x = Random.glucose, group = !is.na(Sex))) +
  geom_histogram(binwidth = 0.1, color="darkblue", fill="lightblue", linetype = "solid") + 
  scale_x_continuous(limits = c(2, 8)) +
  xlab("Random Glucose (mmol/L)") +
  ylab("Participant count") + 
  ggtitle("RG Distribution in Study Population") + 
  facet_grid(. ~Sex, labeller=labeller(Sex = labels)) +
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))

ggsave("RG Distribution by Sex.jpg")

# plotting normalised RG levels (log 10), separated by sex 
ggplot(epi_data[!is.na(epi_data$Sex), ], aes(x = Random.glucose.log, group = !is.na(Sex))) +
  geom_histogram(binwidth = 0.01, color="darkblue", fill="lightblue", linetype = "solid") + 
  scale_x_continuous(limits = c(1, 2.5)) +
  xlab("log10(Random Glucose)") + #check the scale here 
  ylab("Participant count") + 
  ggtitle("RG Distribution in Study Population") + 
  facet_grid(. ~Sex, labeller=labeller(Sex = labels)) + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("Normalised RG Distribution by Sex log10.jpg")

```

The range of RG levels in the study population is approximately normally distributed.

## Body Mass Index (BMI)

Body Mass Index is a measure used to work out whether an individual's weight is healthy. A BMI calculation is performed by dividing an individual's weight (in kg) by their height (in metres) squared (thus, units = kg/m\^2). BMI values are often stratified into different levels, depending on the what "state of health" the value indicates.

```{r Body Mass Index, echo = FALSE}

# plotting BMI in bins of 1.0 increase, no stratification 
ggplot(epi_data[!is.na(epi_data$Sex), ], aes(x = BMI)) +
  geom_histogram(binwidth = 1, color="darkblue", fill="lightblue", linetype = "solid") + 
  scale_x_continuous(limits = c(15, 55)) +
  xlab("BMI (kg/m^2)") +
  ylab("Participant count") + 
  ggtitle("BMI Distribution in Study Population") + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("BMI Distribution.jpg")

# plotting BMI in bins of 1.0 increase, stratified by sex 
# labels for facet grid
labels = c("0" = "Female", "1" = "Male")

ggplot(epi_data[!is.na(epi_data$Sex), ], aes(x = BMI, group = !is.na(Sex))) +
  geom_histogram(binwidth = 1, color="darkblue", fill="lightblue", linetype = "solid") + 
  scale_x_continuous(limits = c(15, 55)) +
  xlab("BMI (kg/m^2)") +
  ylab("Participant count") + 
  ggtitle("BMI Distribution in Study Population") + 
  facet_grid(. ~Sex, labeller=labeller(Sex = labels)) +
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("BMI Distribution by Sex.jpg")

# adding column of BMI categories 
epi_data$BMI.category = cut(epi_data$BMI, breaks = c(10, 18.5, 25, 30, 35, 40, 45), 
                            labels = c("Underweight", "Normal", "Overweight", "Obese I", 
                            "Obese II", "Obese III"), right = T)

# plotting BMI in bins of 5.0 increase (different BMI categories)
ggplot(data = subset(epi_data, !is.na(BMI.category)), aes(x = BMI.category, fill = BMI.category, 
                                                          na.rm = T)) +
  geom_bar() +
  scale_fill_manual(values = c("#85C1E9", "#58D68D", "#F7DC6F", "#F2965D", "#E67E22", "#ED1010"), 
                    name = "BMI Classes", breaks = c("Underweight", "Normal", "Overweight", "Obese I", 
                            "Obese II", "Obese III"), labels = c("< 18.5", "18.5-24.9", "25-29.9",
                                                                 "30-34.9", "35-39.9", "> 40")) + 
  labs(x = "BMI (kg/m^2)", y = "Participant count") +
  scale_y_continuous(expand = expansion(mult = c(0,0.1))) +
  ggtitle("BMI Classification in Study Population") + 
  theme(axis.title.y = element_text(vjust = +5)) +
  theme_bw() + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)), axis.title.x =
          element_text(margin = margin(t = 5)))
ggsave("BMI Distribution Classes.jpg")


```

We observe that the majority of individuals in the study population are in the "overweight" BMI category, with a BMI of 25-29.9 kg/m\^2.

### Summary Statistics

The below calculations of summary statistics are performed in order to provide an overview of key characteristics of the study population. They are calculated for the whole cohort, only for T2D patients, only for PanC patients, and for individuals with both a T2D and PanC diagnosis.

## Summary Statistics of Whole Cohort

```{r Whole Cohort Summary Statistics, echo = TRUE}
# Mean and SD of RG measurements for whole cohort 
mean_RG_all = mean(epi_data$Random.glucose, na.rm = T)
SD_RG_all = sd(epi_data$Random.glucose, na.rm = T)

# Mean and SD of BMI for whole cohort
mean_BMI_all = mean(epi_data$BMI, na.rm = T)
SD_BMI_all = sd(epi_data$BMI, na.rm = T)

# Mean and SD of age for whole cohort
mean_age_all = mean(epi_data$AgeBaseline, na.rm = T)
SD_age_all = sd(epi_data$AgeBaseline, na.rm = T)

# Sex distribution for whole cohort
female_count_all = sum(epi_data$Sex == 0)
male_count_all = sum(epi_data$Sex == 1)

perc_women = sum(epi_data$Sex == 0, na.rm = T)/nrow(epi_data)
perc_men = sum(epi_data$Sex == 1, na.rm = T)/nrow(epi_data)

```

## Summary Statistics for PanC cases

```{r PANC Summary Statistics, echo = TRUE}
# Mean and SD of RG for all participants with a PANC diagnosis
mean_RG_level_PANC = mean(subset(epi_data, epi_data$C25 == 1)$Random.glucose, na.rm = T)
SD_RG_level_PANC = sd(subset(epi_data, epi_data$C25 == 1)$Random.glucose, na.rm = T)

# Mean and SD of BMI for all participants with a PANC diagnosis 
mean_BMI_PANC = mean(subset(epi_data, epi_data$C25 == 1)$BMI, na.rm = T)
SD_BMI_PANC = sd(subset(epi_data, epi_data$C25 == 1)$BMI, na.rm = T)

# Mean and SD of age for all participants with a PANC diagnosis 
mean_age_PANC = mean(subset(epi_data, epi_data$C25 == 1)$AgeBaseline, na.rm = T)
SD_age_PANC = sd(subset(epi_data, epi_data$C25 == 1)$AgeBaseline, na.rm = T)

# Sex distribution for all participants with a PANC diagnosis
female_count_PANC = sum(subset(epi_data, epi_data$C25 ==1)$Sex == 0)
male_count_PANC = sum(subset(epi_data, epi_data$C25 ==1)$Sex == 1)

perc_women_PANC = sum(subset(epi_data, epi_data$C25 ==1)$Sex == 0)/nrow(subset(epi_data,
                  epi_data$C25 == 1))                               
perc_men_PANC = sum(subset(epi_data, epi_data$C25 ==1)$Sex == 1)/nrow(subset(epi_data,
                 epi_data$C25 == 1))
```

## Summary Statistics for T2D cases

```{r T2D Summary Statistics, echo = TRUE}
# Mean and SD of RG for all participants with a T2D diagnosis
mean_RG_level_T2D = mean(subset(epi_data, epi_data$T2D == 1)$Random.glucose, na.rm = T)
SD_RG_level_T2D = sd(subset(epi_data, epi_data$T2D == 1)$Random.glucose, na.rm = T)

# Mean and SD of BMI for all participants with a T2D diagnosis 
mean_BMI_T2D = mean(subset(epi_data, epi_data$T2D == 1)$BMI, na.rm = T)
SD_BMI_T2D = sd(subset(epi_data, epi_data$T2D == 1)$BMI, na.rm = T)

# Mean and SD of age for all participants with a T2D diagnosis 
mean_age_T2D = mean(subset(epi_data, epi_data$T2D == 1)$AgeBaseline, na.rm = T)
SD_age_T2D = sd(subset(epi_data, epi_data$T2D == 1)$AgeBaseline, na.rm = T)

# Sex distribution for all participants with a T2D diagnosis
female_count_T2D = sum(subset(epi_data, epi_data$T2D ==1)$Sex == 0)
male_count_T2D = sum(subset(epi_data, epi_data$T2D ==1)$Sex == 1)

perc_women_T2D = sum(subset(epi_data, epi_data$T2D ==1)$Sex == 0)/nrow(subset(epi_data,
                  epi_data$T2D == 1))                               
perc_men_T2D = sum(subset(epi_data, epi_data$T2D ==1)$Sex == 1)/nrow(subset(epi_data,
                 epi_data$T2D == 1))

```

## Summary Statistics for T2D + PanC cases

```{r T2D + PANC Summary Statistics, echo = TRUE}
mean_RG_level_T2D_PANC = mean(subset(epi_data, epi_data$T2D.PANC == 1)$Random.glucose, na.rm = T)
SD_RG_level_T2D_PANC = sd(subset(epi_data, epi_data$T2D.PANC == 1)$Random.glucose, na.rm = T)

# Mean and SD of BMI for all participants with a PANC diagnosis 
mean_BMI_T2D_PANC = mean(subset(epi_data, epi_data$T2D.PANC == 1)$BMI, na.rm = T)
SD_BMI_T2D_PANC = sd(subset(epi_data, epi_data$T2D.PANC == 1)$BMI, na.rm = T)

# Mean and SD of age for all participants with a PANC diagnosis 
mean_age_T2D_PANC = mean(subset(epi_data, epi_data$T2D.PANC == 1)$AgeBaseline, na.rm = T)
SD_age_T2D_PANC = sd(subset(epi_data, epi_data$T2D.PANC == 1)$AgeBaseline, na.rm = T)

# Sex distribution for all participants with a PANC diagnosis
female_count_T2D_PANC = sum(subset(epi_data, epi_data$T2D.PANC ==1)$Sex == 0)
male_count_T2D_PANC = sum(subset(epi_data, epi_data$T2D.PANC ==1)$Sex == 1)

perc_women_T2D_PANC = sum(subset(epi_data, epi_data$T2D.PANC ==1)$Sex == 0)/nrow(subset(epi_data, epi_data$T2D.PANC == 1))                               
perc_men_T2D_PANC = sum(subset(epi_data, epi_data$T2D.PANC ==1)$Sex == 1)/nrow(subset(epi_data, epi_data$T2D.PANC == 1))
```

## Polygenic Risk Scores (PRS)

Polygenic Risk Scores (PRS) are used to quantify an individual's genetic disease risk, by constructing a sum which is unweighted, or weighted by effect size, of all disease-associated risk alleles present in the individual's genome.

PRS for 1) T2D and 2) PanC have been calculated for each participant in the study population. The below plots show the distribution of the two PRS for the whole cohort. As expected, the PRS are normally distributed.

```{r, echo = FALSE}
# import and merge PANC PRS with epi_data 
panc_prs = fread("~/Desktop/Masters Project/GxE Testing PanC/ukbb_panc_snps_panc_betas_score.profile") %>% mutate(score_norm = scale(SCORESUM))
epi_data_panc_prs = merge(panc_prs, epi_data, by.x = "IID", by.y = "ID_1")

# import and merge T2D PRS with epi_data
t2d_prs = fread("~/Desktop/Masters Project/GxE Testing PanC/ukbb_t2d_weighted_score.profile") %>% mutate(score_norm = scale(SCORESUM))
epi_data_t2d_prs = merge(t2d_prs, epi_data, by.x = "IID", by.y = "ID_1")

# Plot density distribution of PANC PRS (should be normally distributed)
ggplot(epi_data_panc_prs, aes(x = score_norm)) +
  geom_density(color="darkblue", fill="#6F8FAF", linetype = "solid") + 
  scale_x_continuous(limits = c(-3, 3)) +
  xlab("Normalised PRS") +
  ylab("Density") + 
  ggtitle("PANC PRS Distribution") + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("PANC PRS Distribution Density.jpg")

# Plot density distribution of T2D PRS (should be normally distributed)
ggplot(epi_data_t2d_prs, aes(x = score_norm)) +
  geom_density(color="darkblue", fill ="#4682B4", linetype = "solid") + 
  scale_x_continuous(limits = c(-3, 3)) +
  xlab("Normalised PRS") +
  ylab("Density") + 
  ggtitle("T2D PRS Distribution") + 
  theme(axis.title.y = element_text(margin = margin(r = 10, l = 5)))
ggsave("T2D PRS Density Distribution.jpg")
```
